tests/60app-services/02ghost.pl

```diff
@@ -189,3 +189,106 @@
          },
       );
    };
+
+
+my $avatar_url = "http://somewhere/my-pic.jpg";
+
+test "AS can set avatar for ghosted users",
+   requires => [ as_ghost_fixture(), $main::AS_USER[0],
+                 qw( can_get_avatar_url ) ],
+
+   check => sub {
+      my ( $ghost, $as_user ) = @_;
+
+      my $user_id = $ghost->user_id;
+
+      my $http = $as_user->http;
+
+      $http->do_request_json(
+         method => "GET",
+         uri    => "/r0/profile/$user_id/avatar_url",
+      )->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( avatar_url ));
+         assert_eq( $body->{avatar_url}, $avatar_url, 'avatar_url' );
+
+         Future->done(1);
+      });
+   },
+
+   do => sub {
+      my ( $ghost, $as_user ) = @_;
+
+      my $user_id = $ghost->user_id;
+
+      do_request_json_for(
+         $as_user,
+         method => "PUT",
+         uri    => "/r0/profile/$user_id/avatar_url",
+         params => {
+            user_id => $user_id,
+         },
+         content => { avatar_url => $avatar_url },
+      );
+   };
```

Comment:
TBH I prefer to merge check and do, and its what we do elsewhere, but *shrugs*

Reply:
It's certainly not what we do wherever I cargo-culted this from. The nice thing about the separate functions is that it checks that you're testing what you think you're testing by making sure that the check fails before the do, and passes after the do.

Reply:
Ah yes, I guess I meant "what we've done recently since we last changed our minds". I don't really have much against it when it makes sense